packages/blocks/src/api/registration.js

```diff
@@ -24,6 +24,101 @@ import { select, dispatch } from '@wordpress/data';
 import { isValidIcon, normalizeIconObject } from './utils';
 import { DEPRECATED_ENTRY_KEYS } from './constants';
 
+/**
+ * TODO: Editor block attributes object.
+ *
+ * @typedef {Object} WPBlockAttributeOptions
+ *
+ * @property {WPBlockAttributeType} type ...
+ * @property {string} source ...
+ * @property {string} selector ...
+ * @property {string} attribute ...
+ * @property {string} meta Attributes may be obtained from a post’s meta rather than from the block’s representation in saved post content. For this, an attribute is required to specify its corresponding meta key under the meta key
```

Comment:
As minor points of code style convention (not currently enforced at this level of detail):

- Consider to wrap at 80-100 line length ([reference guideline](https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/#spacing)
- Consider to align respective "sections" of a grouping (i.e. horizontally aligned types, names, descriptions within a `@property` grouping) ([reference guideline](https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/#functions), though it should be noted the document in general has fallen somewhat out of date)

```suggestion
 * @property {string}               meta Attributes may be obtained from post
 *                                       meta rather than from the block’s
 *                                       representation in saved post content.
 *                                       For this, an attribute is required to
 *                                       specify its corresponding meta key
 *                                       under the meta key.
```

Reply:
Noted, thanks! This entire block is marked as `TODO` but I will format it accordingly after adding all of the descriptions and correct types.